include/xtensor/xfft.hpp

```diff
@@ -0,0 +1,294 @@
+#ifndef XFFT_HPP
+#define XFFT_HPP
+
+#include "xmath.hpp"
+#include "xcomplex.hpp"
+#include "xutils.hpp"
+#include "xaxis_slice_iterator.hpp"
+#include "xio.hpp"
+#include "xview.hpp"
+
+namespace xt 
+{
+    namespace fft
+    {
+        template<typename T = double, typename E1, typename E2>
+        xt::xarray<std::complex<T>> fftconvolve(E1&& xvec, E2&& yvec, std::ptrdiff_t axis = -1);
```

Comment:
Would be nice to return an `xtensor` is the rank static.

Reply:
It's not static it's a one dimensional transform performed along a specified axis thus the axis parameter.

Reply:
I get that. What I meant is: if the input has a static rank one could output a static rank

Reply:
Oh yeah that's... I ran into some issues here with things being in the header then having to forward declare. Turns out you cannot use auto on a forward declared function. I'll take a look and see if I can make it work with static rank. How much performance does one gain from xtensor over xarray?

Reply:
@tdegeus Unfortunately I think we have hit a limitation of C++. FFT calls fft_convolve and fft_convolve calls fft so there is a circular relationship because their implementation relies on each other. Ideally I would be able to make the return type auto but I cannot do that and declare the function separate from it's definition thus I think we are limited to the more generic xarray<T> return type... Let me know if you see a way to fix it.